components/authz-service/server/v2/project.go

```diff
@@ -367,14 +367,17 @@ func (s *state) UpdateRule(ctx context.Context, req *api.UpdateRuleReq) (*api.Up
 }
 
 func (s *state) GetRule(ctx context.Context, req *api.GetRuleReq) (*api.GetRuleResp, error) {
-	resp, err := s.store.GetRule(ctx, req.Id)
+	resp, err := s.store.GetStagedOrAppliedRule(ctx, req.Id)
 	if err != nil {
 		if err == storage_errors.ErrNotFound {
 			return nil, status.Errorf(codes.NotFound, "could not find rule with ID %q", req.Id)
 		}
 		return nil, status.Errorf(codes.Internal,
 			"error retrieving rule with ID %q: %s", req.Id, err.Error())
 	}
+	if resp.Deleted {
+		return nil, status.Errorf(codes.NotFound, "could not find rule with ID %q", req.Id)
+	}
```

Comment:
Interesting. So we just give them a 404 if the rule is staged to be deleted? Is this correct? What is list going to do with a rule that is staged to be deleted?

Reply:
This behavior comes from our interpretation of [one of the Staging Changes requirements](https://chefio.atlassian.net/browse/A2-591):

> If multiple users are staging changes, each user should be able to see the changes of the other user

users should always see the most recent change to any given rule. when fetching a rule, we return the staged version if it exists in the staged table, even though the staged change hasn't been applied yet. so if the staged version of the rule is marked deleted, we treat it as if it's already been deleted.

Reply:
Presuming that `resp.Deleted` will only be true if the item is marked deleted in the staging table, I would suggest make the message more transparent: "rule with ID %q is staged for deletion".

Reply:
@msorens At first I was onboard with that suggestion, but I think it might create some additional confusion for users at this junction given that while _we_ know it's staged for deletion, they don't need to since 1) It's effectively deleted-- there's nothing they can do about it and 2) there will be no way for them to distinguish staged vs 'applied' in any other context (updated/new, etc). 

In the future, if we do begin to distinguish, I think your error msg makes lots of sense.